controllers/clusterurlmonitor/clusterurlmonitor_supplement.go

```diff
@@ -214,13 +215,27 @@ func (s *ClusterUrlMonitorReconciler) getHypershiftClusterDomain(monitor v1alpha
 	if err != nil {
 		return "", fmt.Errorf("failed to retrieve HostedControlPlane for hosted cluster: %w", err)
 	}
-
 	clusterAnnotation := clusterHCP.Annotations[hcpClusterAnnotation]
 	annotationTokens := strings.Split(clusterAnnotation, "/")
 	if len(annotationTokens) != 2 {
 		return "", fmt.Errorf("invalid annotation for HostedControlPlane '%s': expected <namespace>/<hostedcluster name>, got %s", clusterHCP.Name, clusterAnnotation)
 	}
 
+	endpointAccess := clusterHCP.Spec.Platform.AWS.EndpointAccess
+
+	// In the case of private clusters, the domain will be the ingress of the private router
+	if endpointAccess == hypershiftv1beta1.Private {
+		service := corev1.Service{}
+		err := s.Client.Get(s.Ctx, types.NamespacedName{Name: "private-router", Namespace: monitor.Namespace}, &service)
+		if err != nil {
+			return "", fmt.Errorf("could not retrieve private router in namespace '%s'; Reason: %s", monitor.Namespace, err.Error())
+		}
+
+		ingress := service.Status.LoadBalancer.Ingress[0].Hostname
```

Comment:
During the initial creation of the service, this status field may not exist yet as it takes some time for a Load Balancer to be created in AWS, so we should add a couple `nil` guards around this and return an error if the service is not fully populated yet.

It should be safe to assume that `service.Status` exists, but anything after may not when this code runs (eventually we can assume that it will be populated).